examples/apps/cli/Makefile.am

```diff
@@ -28,28 +28,36 @@
 
 include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
-bin_PROGRAMS                                                      = \
-    ot-cli                                                          \
+bin_PROGRAMS                                                           = \
+    ot-cli                                                               \
     $(NULL)
 
-ot_cli_CPPFLAGS                                                   = \
-    -I$(top_srcdir)/include                                         \
-    -I$(top_srcdir)/examples/cc2538/platform                        \
+ot_cli_CPPFLAGS                                                        = \
+    -I$(top_srcdir)/include                                              \
+    -I$(top_srcdir)/examples/platforms                                   \
     $(NULL)
 
-ot_cli_LDADD                                                      = \
-    $(top_builddir)/src/core/libopenthread.a                        \
-    $(top_builddir)/src/cli/libopenthread-cli.a                     \
-    $(top_builddir)/examples/cc2538/platform/libopenthread-cc2538.a \
-    $(top_builddir)/third_party/mbedtls/libmbedcrypto.a             \
+ot_cli_LDADD                                                           = \
+    $(top_builddir)/src/core/libopenthread.a                             \
+    $(top_builddir)/src/cli/libopenthread-cli.a                          \
+    $(top_builddir)/third_party/mbedtls/libmbedcrypto.a                  \
+    -lstdc++                                                             \
```

Comment:
Perhaps using a build flag to allow a port to specify what stdc++ lib it wants to use would be good.  On many embedded ports, these __cxa calls would be stubbed out or handled by libnano.